HeatStroke/Services/Collisions/CollisionManager.cpp

```diff
@@ -271,5 +274,15 @@ void CollisionManager::CheckCollision(ComponentSphereCollider* p_pCollider1, Com
 		collisionEvent->SetIntParameter("XAligned", xAligned);
 		collisionEvent->SetIntParameter("ZAligned", zAligned);
 		EventManager::Instance()->TriggerEvent(collisionEvent);
+
+		CollisionEffect(contactPoint);
 	}
+}
+
+void CollisionManager::CollisionEffect(glm::vec3 p_vPosition)
+{
+	GameObject* pEffect = m_pGameObjectManager->CreateGameObject("CS483/CS483/Kartaclysm/Data/Racer/collision_effect.xml", "CollisionEffect" + std::to_string(m_iEffectIndex));
+	m_iEffectIndex++;
```

Comment:
Switching to an unsigned int would mean we can produce twice the number of particle GUIDs without overflow